File: Python/TotalWavinessofNumbersInRange_I.py
class TotalWavinessofNumbersInRange_I(object):
    def totalWaviness(self, num1, num2):
        total = 0
        for n in range(num1, num2 + 1):
            s = str(n)
            for i in range(1, len(s) - 1):
                if (s[i] > s[i-1] and s[i] > s[i+1]) or \
                   (s[i] < s[i-1] and s[i] < s[i+1]):
                    total += 1
        return total
    

twnir = TotalWavinessofNumbersInRange_I()
num1 = 120
num2 = 130
result = twnir.totalWaviness(num1, num2)
print("Total Waviness:", result)

File: Java/TotalWavinessofNumbersInRange_I.java
class TWNIR_I {
    public int totalWaviness(int num1, int num2) {
        int total = 0;
        for (int n = num1; n <= num2; n++) {
            String s = Integer.toString(n);
            int len = s.length();
            for (int i = 1; i < len - 1; i++) {
                if ((s.charAt(i) > s.charAt(i-1) && s.charAt(i) > s.charAt(i+1)) ||
                    (s.charAt(i) < s.charAt(i-1) && s.charAt(i) < s.charAt(i+1)))
                    total++;
            }
        }
        return total;
    }
}

public class TotalWavinessofNumbersInRange_I {
    public static void main(String[] args) {
        TWNIR_I twnir = new TWNIR_I();
        int num1 = 120, num2 = 130;
        int result = twnir.totalWaviness(num1, num2);
        System.out.println("Total Waviness: " + result);
    }
}
